### PR TITLE
update ldapprobe to include count of objects

### PR DESCRIPTION
**Pull request by ctrl78**
add new function to count ldap objects

### CODE REVIEW

File: prober/ldap.go [hunk 1]
```diff
@@ -87,5 +95,34 @@ func ProbeLDAP(ctx context.Context, target string, module config.Module, registy
...
 	}
 	probeLDAPStatusCode.Set(0)
+
+	searchRequest := ldap.NewSearchRequest(
+		module.LDAP.Query.DN, // The base dn to search
+		config.LDAPScopes[module.LDAP.Query.Scope], ldap.NeverDerefAliases, -1, 0, false,
+		module.LDAP.Query.Filter,
+		//"(&(objectClass=organizationalPerson))", // The filter to apply
+		module.LDAP.Query.Attributes,
+		//[]string{"dn", "cn"}, // A list attributes to retrieve
```
Comment (uepoch) [inline prober/ldap.go]: Please remove comments

File: prober/ldap.go [hunk 2]
```diff
@@ -87,5 +95,34 @@ func ProbeLDAP(ctx context.Context, target string, module config.Module, registy
 		return false
 	}
 	probeLDAPStatusCode.Set(0)
+
+	searchRequest := ldap.NewSearchRequest(
+		module.LDAP.Query.DN, // The base dn to search
+		config.LDAPScopes[module.LDAP.Query.Scope], ldap.NeverDerefAliases, -1, 0, false,
+		module.LDAP.Query.Filter,
+		//"(&(objectClass=organizationalPerson))", // The filter to apply
```
Comment (uepoch) [inline prober/ldap.go]: Please remove comments

File: config/config.go [hunk 3]
```diff
@@ -134,8 +135,19 @@ type LDAPBind struct {
 	XXX map[string]interface{} `yaml:",inline"`
 }
 
+type LDAPQuery struct {
+	DN         string
+	Filter     string
+	Scope      string // Defaults to "(objectClass=*)".
```
Comment (uepoch) [inline config/config.go]: Wrong Comment, it's the filter not the scope

**Review by uepoch — CHANGES_REQUESTED** [review]
What happens if the Query object is nil ?

File: prober/ldap.go (hunk 1 above)
Comment (ctrl78) [reply prober/ldap.go]: done

File: prober/ldap.go (hunk 2 above)
Comment (ctrl78) [reply prober/ldap.go]: done

File: config/config.go (hunk 3 above)
Comment (ctrl78) [reply config/config.go]: indeed removed

Comment (ctrl78) [reply config/config.go]: ldap lib have default value -> // Defaults to "(objectClass=*)".

**Comment by ctrl78** [comment]
About having empty values checks are in / UnmarshalYAML implements the yaml.Unmarshaler interface.
func (s *LDAPQuery) UnmarshalYAML(unmarshal func(interface{}) error) error {

	type plain LDAPQuery
	if err := unmarshal((*plain)(s)); err != nil {
		return err
	}

	if err := checkOverflow(s.XXX, "ldap query"); err != nil {
		return err
	}

	if _, err := ldap.CompileFilter(s.Filter); err != nil && s.Filter != "" {
		return fmt.Errorf("Invalid filter detected: %s", s.Filter)
	}

	if _, err := ldap.ParseDN(s.DN); err != nil {
		return fmt.Errorf("Invalid DN detected: %s", s.DN)
	}

	if _, ok := LDAPScopes[s.Scope]; !ok {
		if s.Scope == "" {
			s.Scope = "one"
		} else {
			return fmt.Errorf("Unknown scope type: %s", s.Scope)
		}
	}
	return nil
}

**Comment by uepoch** [comment]
Having no query isn't an error, you could check only the binding of the LDAP if you wanted to

**Pull request merged**